compute_sdk/tests/unit/test_executor.py

```diff
@@ -75,7 +75,11 @@ def __init__(self, *args, **kwargs):
 
     def start(self) -> None:
         super().start()
+
+        # important for tests to ensure the `.run()` has had a chance to be invoked
+        # before the tests poke at the internals.
         try_assert(lambda: self._connection is not None)
+        try_assert(lambda: self._channel is not None)
```

Comment:
Since a channel requires a connection, we could drop the first check.

Reply:
*nod; I thought about that.  But that presumes that the two are structurally connected&nbsp;&mdash;&nbsp;which they are in the "real code," but not here in the tests:

```python
    def run(self):
        self._connection = mock.MagicMock()
        self._channel = mock.MagicMock()
        self._time_to_stop_mock.wait()
```
They are clearly in the same function, but there is no data structure tie between them.